src/transforms/add_fields.rs

```diff
@@ -97,7 +97,14 @@ impl Transform for AddFields {
                 .into(),
                 TemplateOrValue::Value(v) => v,
             };
-            event.as_mut_log().insert(key, value);
+            let old_val = event.as_mut_log().insert(&key, value);
+            if old_val.is_some() {
```

Comment:
Since you don't actually use the old value, this seems simpler:
`if let Some(_) = event.as_mut_log().insert(&key, value) {`
or if you prefer the `is_some`:
`if event.as_mut_log().insert(&key, value).is_some() {`